File: scrooge-generator/src/main/resources/javagen/readField.java
case {{id}}: /* {{fieldName}} */
  switch (_field.type) {
    case TType.{{constType}}:
      {{fieldType}} {{valueVariableName}};
      {{>readValue}}
      {{fieldName}} = {{valueVariableName}};
      break;
    default:
      TProtocolUtil.skip(_iprot, _field.type);
  }

File: scrooge-generator/src/main/resources/cocoagen/readField.m
case {{id}}:
    if (fieldType == TType_{{wireConstType}}) {
        {{fieldType}}{{#readWriteInfo}}{{#isStruct}}*{{/isStruct}}{{/readWriteInfo}} {{valueVariableName}};
        {{>readValue}}
        [self set{{FieldName}}:{{valueVariableName}}];
    } else {
        NSLog(@"%s: field ID %i has unexpected type %i.  Skipping.", __PRETTY_FUNCTION__, fieldID, fieldType);
        [TProtocolUtil skipType:fieldType onProtocol:inProtocol];
    }
    break;
